tools/flang1/flang1exe/semfunc.c

```diff
@@ -1247,8 +1248,39 @@ func_call2(SST *stktop, ITEM *list, int flag)
     goto exit_;
   }
   ii = 0;
-  /* before processing arguments, add derived type return values if needed */
-  argt = mk_argt(argt_count); /* mk_argt stuffs away count */
+  /* As for assumed-length character function, get a temporary with DT_DEFERCHAR, SC_BASED here */
+  if ((DTY(dtype) == TY_CHAR) && ADJLENG(func_sptr)) {
+    int functmp;
+    int funcast, alloc_ast, len_ast;
+    int alloc_stmt, len_stmt;
+
+    functmp = getcctmp_sc('d', sem.dtemps++, ST_VAR, DT_DEFERCHAR, SC_BASED);
+    funcast = mk_id(functmp);
```

Comment:
`funcast` should be renamed to `functmp_ast`, to indicate their relationship clearly.